File: src/test/java/com/intellij/plugin/powershell/lang/PowerShellParserTest.java
package com.intellij.plugin.powershell.lang;

import com.intellij.plugin.powershell.lang.parser.PowerShellParserDefinition;
import com.intellij.testFramework.ParsingTestCase;

import java.io.File;

/**
 * Andrey 06/09/17.
 */
public class PowerShellParserTest extends ParsingTestCase {
  private static final String TEST_DATA_PATH = "src" + File.separatorChar + "test" + File.separatorChar + "resources" + File.separatorChar +
      "testData";

  public PowerShellParserTest() {
    super("parser", "ps1", new PowerShellParserDefinition());
  }

  @Override
  protected String getTestDataPath() { return TEST_DATA_PATH; }

  @Override
  protected boolean includeRanges() { return true; }

  public void testCallInvocationOperator() throws Exception { doTest(true); }

  public void testVerbatimCommandArgument() throws Exception { doTest(true); }

  public void testPathExpression() throws Exception { doTest(true); }

  public void testCommandCallExpression() throws Exception { doTest(true); }

  public void testStdCmdlets() throws Exception { doTest(true); }

  public void testMemberElementAccess() throws Exception { doTest(true); }

  public void testOperators() throws Exception { doTest(true); }

  public void testLabeledStatement() throws Exception { doTest(true); }

  public void testStatements() throws Exception { doTest(true); }

  public void testArrayStatement() throws Exception { doTest(true); }

  public void testAttributes() throws Exception { doTest(true); }

  public void testClosures() throws Exception { doTest(true); }

  public void testModules() throws Exception { doTest(true); }

  public void testHashTables() throws Exception { doTest(true); }

  public void testTypeLiteral() throws Exception { doTest(true); }

}

File: src/test/kotlin/com/intellij/plugin/powershell/lang/PowerShellParserTest.kt
package com.intellij.plugin.powershell.lang

import com.intellij.lang.LanguageBraceMatching
import com.intellij.plugin.powershell.ide.editor.highlighting.PowerShellPairedBraceMatcher
import com.intellij.plugin.powershell.lang.parser.PowerShellParserDefinition
import com.intellij.testFramework.ParsingTestCase

class PowerShellParserTest : ParsingTestCase("parser", "ps1", PowerShellParserDefinition()) {

  override fun setUp() {
    super.setUp()
    addExplicitExtension(LanguageBraceMatching.INSTANCE, PowerShellLanguage.INSTANCE, PowerShellPairedBraceMatcher())
  }

  override fun getTestDataPath() =  "src/test/resources/testData"
  override fun includeRanges() = true

  fun testCallInvocationOperator() { doTest(true) }
  fun testVerbatimCommandArgument() { doTest(true) }
  fun testPathExpression() { doTest(true) }
  fun testCommandCallExpression() { doTest(true) }
  fun testStdCmdlets() { doTest(true) }
  fun testMemberElementAccess() { doTest(true) }
  fun testTypeMemberAccess() { doTest(true) }
  fun testOperators() { doTest(true) }
  fun testNullCoalesceOperator() { doTest(true) }
  fun testLabeledStatement() { doTest(true) }
  fun testStatements() { doTest(true) }
  fun testFunctionMultilineDefinition() { doTest(true) }
  fun testDesiredSateConfiguration() { doTest(true) }
  fun testArrayStatement() { doTest(true) }
  fun testAttributes() { doTest(true) }
  fun testClosures() { doTest(true) }
  fun testModules() { doTest(true) }
  fun testHashTables() { doTest(true) }
  fun testExpandableString() { doTest(true) }
  fun testVerbatimString() { doTest(true) }
  fun testTypeLiteral() { doTest(true) }
  fun testClassDeclaration() { doTest(true) }
  fun testEnumDeclaration() { doTest(true) }
  fun testParamBlock() { doTest(true) }
  fun testComment() { doTest(true) }
  fun testNewLineThenPipeOperator() { doTest(true) }
}
